modules/nativescript/nativescript.cpp

```diff
@@ -754,7 +784,16 @@ NativeScriptInstance::~NativeScriptInstance() {
 	script_data->destroy_func.destroy_func((godot_object *)owner, script_data->destroy_func.method_data, userdata);
 
 	if (owner) {
+
+#ifndef NO_THREADS
+		script->owners_lock->lock();
+#endif
+
 		script->instance_owners.erase(owner);
+
+#ifndef NO_THREADS
+		script->owners_lock->lock();
+#endif
```

Comment:
Yes, but I feel like it's a bit overkill to lock it every time. After all the map is supposed to be filled once and then only accessed read only. Maybe you have an idea how to do it nicely without much unnecessary locking.